docs/docs-beta/docs/guides/project-structure.md

```diff
@@ -0,0 +1,214 @@
+---
+title: "How to structure your Dagster project"
+---
+
+# How to structure your Dagster project
+
+:::note
+Refer to the project scaffolding tutorial to learn how to create a new Dagster project.
+:::
+
+When it comes to organizing your project, there are many possibilities in how files and directories can be structured. Factors that can influence this includes the number of teams working on your data platform, and whether you want to structure files by Dagster object type, by technology, or by the overarching data concept.
+
+When structuring complex projects, some important considerations are as follows:
+
+1. The location to add new features is intuitive
+2. Relevant code should be easy to find regardless of familiarity with related business logic
+3. The project can be refactored and organized over time without premature optimizations
+
+As your experience with Dagster grows, certain aspects of this guide might no longer apply to your use cases, and you may want to change the structure to adapt to your business needs.
+
+## What you'll learn
+
+- How to restructure your project as your code base grows
+- How to combine multiple `Definitions` objects using `Definitions.merge`
+- How to integrate external projects like dbt or Sling into your codebase
+
+## Your initial project structure
+
+When you first scaffold your project using the Dagster command-line tool, an `assets.py` and `definitions.py` are created in the root of your project.
+
+```sh
+$ dagster project scaffold --name example-dagster-project
+```
+
+```
+.
+├── README.md
+├── example_dagster_project
+│   ├── __init__.py
+│   ├── assets.py
+│   └── definitions.py
+├── example_dagster_project_tests
+│   ├── __init__.py
+│   └── test_assets.py
+├── pyproject.toml
+├── setup.cfg
+└── setup.py
+```
+
+This is a great structure as you are first getting started, however, as you begin to introduce more assets, jobs, resources, sensors, and utility code, you may find that your Python files are growing too large to manage.
+
+## Restructure your project
+
+There are several paradigms in which you can structure your project. Choosing one of these structures is often personal preference, and influenced by how you and your team members operate. This guide will outline three possible project structures:
+
+1. [Structured by Dagster object](#structured-by-object-type)
+2. [Structured by technology](#structured-by-technology)
+3. [Structured by concept](#structured-by-oncept)
+
+### Structured by Dagster object
```

Comment:
I think we should move to #3 and specifically say we do not recommend this approach as it does not scale well and can make it hard to keep related code close together.

Reply:
Removed 1 to emphasize options 2 and 3.